pyvrp/Result.py

```diff
@@ -40,16 +42,23 @@ def __post_init__(self):
         if self.runtime < 0:
             raise ValueError("Negative runtime not understood.")
 
-    def cost(self) -> float:
+    def cost(self, penalty_manager: Optional[PenaltyManager] = None) -> float:
```

Comment:
@leonlan @wouterkool: does it make sense to have a default here? I'm not sure right now. I think I'd rather have that we force a user to be explicit about this, since I cannot really answer clearly what the default here should be.

A middle ground that make sense to me would be the following:
- If the individual is feasible, return the cost with a default PenaltyManager. The penalty terms are zero, so this is just the distance.
- If the individual is infeasible, refuse the temptation to guess what the cost should be. Just raise a ValueError and explicitly force a user to provide a PenaltyManager instance.

Reply:
I agree with this middle ground, although I want to note that it means that the Result uses implicit knowledge on how the cost is computed for feasible individuals.

If we are ok with this in general, I propose to also have a default `.cost()` method in individual, perhaps by separating `.cost()` and `.penalizedCost(PenaltyManager pm)` where the first returns an exception for `infeasible` individuals.

We plan to remove `cost` completely (to `CostEvaluator`) (see #191), but in this case we could also keep the default `cost` and only move the `penalizedCost`. This would also allow us to print the cost in the representation for feasible individuals.

Reply:
Added comment to #100 to discuss this.